### PR TITLE
Add docment to Datasets

### PR DESCRIPTION
**Pull request by malligaraj**
Implement `Datasets` task in issue #8

### CODE REVIEW

File: fastai/data/core.py [hunk 1]
```diff
@@ -323,7 +323,14 @@ def show_at(o, idx, **kwargs):
 @delegates(TfmdLists)
 class Datasets(FilteredBase):
     "A dataset that creates a tuple from each `tfms`"
-    def __init__(self, items=None, tfms=None, tls=None, n_inp=None, dl_type=None, **kwargs):
+    def __init__(self,
+        items=None, # List of items to create `Datasets`
+        tfms=None, # List of `Transform`(s) or `Pipeline` to apply
+        tls=None, # A `TfmdLists` object
```
Comment (kevinbird15) [inline fastai/data/core.py:329]:
```suggestion
        tls:TfmdLists=None, # If None, self.tls is generated from `items` and `tfms`
```

File: fastai/data/core.py [hunk 2]
```diff
@@ -323,7 +323,14 @@ def show_at(o, idx, **kwargs):
 @delegates(TfmdLists)
 class Datasets(FilteredBase):
     "A dataset that creates a tuple from each `tfms`"
-    def __init__(self, items=None, tfms=None, tls=None, n_inp=None, dl_type=None, **kwargs):
+    def __init__(self,
+        items=None, # List of items to create `Datasets`
+        tfms=None, # List of `Transform`(s) or `Pipeline` to apply
```
Comment (kevinbird15) [inline fastai/data/core.py:328]:
```suggestion
        tfms:(list,Pipeline)=None, # List of `Transform`(s) or `Pipeline` to apply
```

Comment (kevinbird15) [inline fastai/data/core.py:328]:
```suggestion
        tfms:list=None, # List of `Transform`(s) or `Pipeline` to apply
```

File: fastai/data/core.py [hunk 3]
```diff
@@ -323,7 +323,14 @@ def show_at(o, idx, **kwargs):
 @delegates(TfmdLists)
 class Datasets(FilteredBase):
     "A dataset that creates a tuple from each `tfms`"
-    def __init__(self, items=None, tfms=None, tls=None, n_inp=None, dl_type=None, **kwargs):
+    def __init__(self,
+        items=None, # List of items to create `Datasets`
```
Comment (kevinbird15) [inline fastai/data/core.py:327]:
```suggestion
        items:list=None, # List of items to create `Datasets`
```

Comment (kevinbird15) [inline fastai/data/core.py:327]:
```suggestion
        items:list=None, # List of items to create `Datasets`
```

**Review by kevinbird15 — CHANGES_REQUESTED** [review]
A few small changes, but overall looking really good

File: fastai/data/core.py [hunk 4]
```diff
@@ -323,7 +323,14 @@ def show_at(o, idx, **kwargs):
 @delegates(TfmdLists)
 class Datasets(FilteredBase):
     "A dataset that creates a tuple from each `tfms`"
-    def __init__(self, items=None, tfms=None, tls=None, n_inp=None, dl_type=None, **kwargs):
+    def __init__(self,
+        items:list=None, # List of items to create `Datasets`
+        tfms:(list,Pipeline)=None, # List of `Transform`(s) or `Pipeline` to apply
+        tls:TfmdLists=None, # If None, `self.tls` is generated from `items` and `tfms`
+        n_inp:int=None, # Number of elements in `Datasets` tuple that should be considered part of input
+        dl_type=None, # Type of `DataLoader`
```
Comment (marii-moe) [inline fastai/data/core.py:331]:
```suggestion
        dl_type=None, # Default type of `DataLoader` used when function `FilteredBase.dataloaders` is called. 
```
This kind of gives more of an idea on what dl_type affects.

**Review by marii-moe — CHANGES_REQUESTED** [review]

**Review by kevinbird15 — APPROVED** [review]
Looks good to me

**Review by marii-moe — APPROVED** [review]
LGTM

**Pull request merged**